libalpm/hooks/90-mkinitcpio-install.hook

```diff
@@ -0,0 +1,17 @@
+[Trigger]
+Type = Package
+Operation = Install
+Operation = Upgrade
+Target = mkinitcpio*
```

Comment:
This is wrong and fails to implement feature compatibility with the current hook, which triggers on `usr/lib/initcpio/*`.

The old hook would trigger on the mkinitcpio package, any packages which were named mkinitcpio-*, and any packages which were named perfectly ordinary innocent things that had nothing to do with mkinitcpio, but also installed initcpio files.

The new hook only supports packages which ship their initcpio files in a split package named mkinitcpio-*. Here's the list of packages affected, most of which would be broken:

```
$ pkgfile -vd /usr/lib/initcpio/
eschwartz/zfs-utils 0.8.2-2           	/usr/lib/initcpio/
core/btrfs-progs 5.2.2-1              	/usr/lib/initcpio/
core/cryptsetup 2.2.1-1               	/usr/lib/initcpio/
core/device-mapper 2.02.186-2         	/usr/lib/initcpio/
core/dmraid 1.0.0.rc16.3-11           	/usr/lib/initcpio/
core/lvm2 2.02.186-2                  	/usr/lib/initcpio/
core/mdadm 4.1-1                      	/usr/lib/initcpio/
core/mkinitcpio 25-2                  	/usr/lib/initcpio/
core/mkinitcpio-busybox 1.30.1-1      	/usr/lib/initcpio/
core/mkinitcpio-nfs-utils 0.3-6       	/usr/lib/initcpio/
core/systemd 243.51-1                 	/usr/lib/initcpio/
extra/archboot 2019.03-1              	/usr/lib/initcpio/
extra/archiso 42-1                    	/usr/lib/initcpio/
extra/ostree 2019.3-3                 	/usr/lib/initcpio/
community/mkinitcpio-dropbear 0.0.3-5 	/usr/lib/initcpio/
community/mkinitcpio-netconf 0.0.5-1  	/usr/lib/initcpio/
community/mkinitcpio-tinyssh 0.0.4-3  	/usr/lib/initcpio/
community/mkinitcpio-utils 0.0.3-4    	/usr/lib/initcpio/
community/pacredir 0.4.2-1            	/usr/lib/initcpio/
```

Reply:
This hook takes care of rebuild the initramfs when any mkinitcpio* package is updated, so, all packages starting with mkinitcpio- on this list are not affected, and I've tested this. I don't like all these packages touch /usr/lib/initcpio, but I guess we have to live with that. It's easy enough to change the hook to trigger on usr/lib/initcpio/*.

Reply:
I have implemented this with 0791321ed88bc480b193889c26cf4c356f1c78f7. The hook triggers on any change to /usr/lib/initcpio.